### ISSUE TITLE
Issues with behind-the-scenes, surprising variable pre-processing and ranger package for random forests

### ISSUE THREAD

**Issue by apreshill** [tidymodels/tune#151]
I consistently run into this problem when using the `ranger` package to run a random forest model and I have a feature that is a factor and happens to have spaces in the cell values themselves.

As you can see, the error messages from `fit_resamples()` and `last_fit()` aren't terribly helpful. I'm intentionally using only a formula here, and it seems surprising how difficult this is to get working with anything but a recipe. This is counter-intuitive because I'm explicitly *not* dummifying this variable, but under the hood I believe that `hardhat` is doing this for me. 

For teaching, given these intricacies, I'm using this function instead (also renamed as we are teaching `last_fit()` first):

```
fit_split <- function(formula, model, split, ...) {
  wf <- workflows::add_model(workflows::add_formula(workflows::workflow(), formula, blueprint = hardhat::default_formula_blueprint(indicators = FALSE)), model)
  tune::last_fit(wf, split, ...)
}
```

``` r
library(tidymodels)
#> Registered S3 method overwritten by 'xts':
#>   method     from
#>   as.zoo.xts zoo
#> ── Attaching packages ───────────────────────────── tidymodels 0.0.3.9000 ──
#> ✓ broom     0.5.3          ✓ purrr     0.3.3     
#> ✓ dials     0.0.4          ✓ recipes   0.1.9     
#> ✓ discrim   0.0.1          ✓ rsample   0.0.5     
#> ✓ dplyr     0.8.3          ✓ tibble    2.1.3     
#> ✓ ggplot2   3.2.1          ✓ tune      0.0.1     
#> ✓ infer     0.5.1          ✓ yardstick 0.0.4     
#> ✓ parsnip   0.0.4.9000
#> ── Conflicts ───────────────────────────────────── tidymodels_conflicts() ──
#> x purrr::discard()    masks scales::discard()
#> x dplyr::filter()     masks stats::filter()
#> x dplyr::lag()        masks stats::lag()
#> x ggplot2::margin()   masks dials::margin()
#> x recipes::step()     masks stats::step()
#> x recipes::yj_trans() masks scales::yj_trans()
library(tune)
library(modeldata)
library(workflows)

data(stackoverflow)

so_split <- initial_split(stackoverflow)
so_train <- training(so_split)

so_fold <- mc_cv(so_train, times = 1)

bag_spec <- 
  rand_forest() %>% 
  set_engine("ranger") %>% 
  set_mode("classification")

# if I use ranger with just a formula, all is fail
# b/c Country has spaces that end up in the dummy variables that get created
# which ranger then trips on
fit_resamples(
  Remote ~ Country,
  model = bag_spec, 
  resamples = so_fold 
)
#> x Resample1: model: Error in parse.formula(formula, data, env = parent.frame()): Erro...
#> Warning: All models failed in [fit_resamples()]. See the `.notes` column.
#> # # Monte Carlo cross-validation (0.75/0.25) with 1 resamples  
#> # A tibble: 1 x 4
#>   splits            id        .metrics .notes          
#> * <list>            <chr>     <list>   <list>          
#> 1 <split [3.1K/1K]> Resample1 <NULL>   <tibble [1 × 1]>

last_fit(
  Remote ~ Country,
  model = bag_spec, 
  split = so_split
)
#> x Resample1: model: Error in parse.formula(formula, data, env = parent.frame()): Erro...
#> Warning: All models failed in [fit_resamples()]. See the `.notes` column.
#> # # Monte Carlo cross-validation (0.75/0.25) with 1 resamples  
#> # A tibble: 1 x 5
#>   splits              id               .metrics .notes           .predictions
#> * <list>              <chr>            <list>   <list>           <list>      
#> 1 <split [4.2K/1.4K]> train/test split <NULL>   <tibble [1 × 1]> <NULL>

# if I wrap with a workflow, all is still fail
bag_wf <- 
  workflow() %>%
  add_formula(Remote ~ Country) %>% 
  add_model(bag_spec)

fit_resamples(
  bag_wf,
  resamples = so_fold 
)
#> x Resample1: model: Error in parse.formula(formula, data, env = parent.frame()): Erro...
#> Warning: All models failed in [fit_resamples()]. See the `.notes` column.
#> # # Monte Carlo cross-validation (0.75/0.25) with 1 resamples  
#> # A tibble: 1 x 4
#>   splits            id        .metrics .notes          
#> * <list>            <chr>     <list>   <list>          
#> 1 <split [3.1K/1K]> Resample1 <NULL>   <tibble [1 × 1]>

last_fit(
  bag_wf,
  split = so_split
)
#> x Resample1: model: Error in parse.formula(formula, data, env = parent.frame()): Erro...
#> Warning: All models failed in [fit_resamples()]. See the `.notes` column.
#> # # Monte Carlo cross-validation (0.75/0.25) with 1 resamples  
#> # A tibble: 1 x 5
#>   splits              id               .metrics .notes           .predictions
#> * <list>              <chr>            <list>   <list>           <list>      
#> 1 <split [4.2K/1.4K]> train/test split <NULL>   <tibble [1 × 1]> <NULL>

# this works, but is very hard to figure out
bp_wf <- 
  workflow() %>%
  add_formula(Remote ~ Country, 
              blueprint = hardhat::default_formula_blueprint(indicators = FALSE)) %>% 
  add_model(bag_spec)

fit_resamples(
  bp_wf,
  resamples = so_fold 
)
#> # # Monte Carlo cross-validation (0.75/0.25) with 1 resamples  
#> # A tibble: 1 x 4
#>   splits            id        .metrics         .notes          
#> * <list>            <chr>     <list>           <list>          
#> 1 <split [3.1K/1K]> Resample1 <tibble [2 × 3]> <tibble [0 × 1]>

last_fit(
  bp_wf,
  split = so_split
)
#> # # Monte Carlo cross-validation (0.75/0.25) with 1 resamples  
#> # A tibble: 1 x 6
#>   splits        id          .metrics      .notes      .predictions     .workflow
#> * <list>        <chr>       <list>        <list>      <list>           <list>   
#> 1 <split [4.2K… train/test… <tibble [2 ×… <tibble [0… <tibble [1,398 … <workflo…

# this also works, but counter-intuitive to need to 
# *un*recipe processing using recipes
so_rec <- 
  recipe(Remote ~ Country, data = so_train) 

fit_resamples(
  so_rec,
  model = bag_spec,
  resamples = so_fold 
)
#> # # Monte Carlo cross-validation (0.75/0.25) with 1 resamples  
#> # A tibble: 1 x 4
#>   splits            id        .metrics         .notes          
#> * <list>            <chr>     <list>           <list>          
#> 1 <split [3.1K/1K]> Resample1 <tibble [2 × 3]> <tibble [0 × 1]>

last_fit(
  so_rec,
  model = bag_spec,
  split = so_split
)
#> # # Monte Carlo cross-validation (0.75/0.25) with 1 resamples  
#> # A tibble: 1 x 6
#>   splits        id          .metrics      .notes      .predictions     .workflow
#> * <list>        <chr>       <list>        <list>      <list>           <list>   
#> 1 <split [4.2K… train/test… <tibble [2 ×… <tibble [0… <tibble [1,398 … <workflo…
```

<sup>Created on 2020-01-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.1 (2019-07-05)
#>  os       macOS Mojave 10.14.5        
#>  system   x86_64, darwin15.6.0        
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/Los_Angeles         
#>  date     2020-01-15                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package       * version    date       lib
#>  assertthat      0.2.1      2019-03-21 [1]
#>  backports       1.1.5      2019-10-02 [1]
#>  base64enc       0.1-3      2015-07-28 [1]
#>  bayesplot       1.7.1      2019-12-01 [1]
#>  boot            1.3-22     2019-04-02 [1]
#>  broom         * 0.5.3      2019-12-14 [1]
#>  callr           3.4.0      2019-12-09 [1]
#>  class           7.3-15     2019-01-01 [1]
#>  cli             2.0.0.9000 2019-12-09 [1]
#>  codetools       0.2-16     2018-12-24 [1]
#>  colorspace      1.4-1      2019-03-18 [1]
#>  colourpicker    1.0        2017-09-27 [1]
#>  crayon          1.3.4      2017-09-16 [1]
#>  crosstalk       1.0.0      2016-12-21 [1]
#>  desc            1.2.0      2018-05-01 [1]
#>  devtools        2.2.1      2019-09-24 [1]
#>  dials         * 0.0.4      2019-12-02 [1]
#>  DiceDesign      1.8-1      2019-07-31 [1]
#>  digest          0.6.23     2019-11-23 [1]
#>  discrim       * 0.0.1      2019-10-11 [1]
#>  dplyr         * 0.8.3      2019-07-04 [1]
#>  DT              0.10       2019-11-12 [1]
#>  dygraphs        1.1.1.6    2018-07-11 [1]
#>  ellipsis        0.3.0      2019-09-20 [1]
#>  evaluate        0.14       2019-05-28 [1]
#>  fansi           0.4.1      2020-01-08 [1]
#>  fastmap         1.0.1      2019-10-08 [1]
#>  foreach         1.4.7      2019-07-27 [1]
#>  fs              1.3.1      2019-05-06 [1]
#>  furrr           0.1.0      2018-05-16 [1]
#>  future          1.15.1     2019-11-25 [1]
#>  generics        0.0.2      2018-11-29 [1]
#>  ggplot2       * 3.2.1      2019-08-10 [1]
#>  ggridges        0.5.1      2018-09-27 [1]
#>  globals         0.12.5     2019-12-07 [1]
#>  glue            1.3.1.9000 2019-12-09 [1]
#>  gower           0.2.1      2019-05-14 [1]
#>  GPfit           1.0-8      2019-02-08 [1]
#>  gridExtra       2.3        2017-09-09 [1]
#>  gtable          0.3.0      2019-03-25 [1]
#>  gtools          3.8.1      2018-06-26 [1]
#>  hardhat         0.1.0.9000 2019-12-20 [1]
#>  highr           0.8        2019-03-20 [1]
#>  htmltools       0.4.0      2019-10-04 [1]
#>  htmlwidgets     1.5.1      2019-10-08 [1]
#>  httpuv          1.5.2      2019-09-11 [1]
#>  igraph          1.2.4.2    2019-11-27 [1]
#>  infer         * 0.5.1      2019-11-19 [1]
#>  inline          0.3.15     2018-05-18 [1]
#>  ipred           0.9-9      2019-04-28 [1]
#>  iterators       1.0.12     2019-07-26 [1]
#>  janeaustenr     0.1.5      2017-06-10 [1]
#>  knitr           1.26.1     2019-12-22 [1]
#>  later           1.0.0      2019-10-04 [1]
#>  lattice         0.20-38    2018-11-04 [1]
#>  lava            1.6.6      2019-08-01 [1]
#>  lazyeval        0.2.2      2019-03-15 [1]
#>  lhs             1.0.1      2019-02-03 [1]
#>  lifecycle       0.1.0      2019-08-01 [1]
#>  listenv         0.8.0      2019-12-05 [1]
#>  lme4            1.1-21     2019-03-05 [1]
#>  loo             2.1.0      2019-03-13 [1]
#>  lubridate       1.7.4      2018-04-11 [1]
#>  magrittr        1.5        2014-11-22 [1]
#>  markdown        1.1        2019-08-07 [1]
#>  MASS            7.3-51.4   2019-03-31 [1]
#>  Matrix          1.2-18     2019-11-27 [1]
#>  matrixStats     0.55.0     2019-09-07 [1]
#>  memoise         1.1.0      2017-04-21 [1]
#>  mime            0.8        2019-12-19 [1]
#>  miniUI          0.1.1.1    2018-05-18 [1]
#>  minqa           1.2.4      2014-10-09 [1]
#>  modeldata     * 0.0.1      2019-12-19 [1]
#>  munsell         0.5.0      2018-06-12 [1]
#>  nlme            3.1-140    2019-05-12 [1]
#>  nloptr          1.2.1      2018-10-03 [1]
#>  nnet            7.3-12     2016-02-02 [1]
#>  parsnip       * 0.0.4.9000 2019-12-09 [1]
#>  pillar          1.4.3      2019-12-20 [1]
#>  pkgbuild        1.0.6      2019-10-09 [1]
#>  pkgconfig       2.0.3      2019-09-22 [1]
#>  pkgload         1.0.2      2018-10-29 [1]
#>  plyr            1.8.5      2019-12-10 [1]
#>  prettyunits     1.1.0      2020-01-09 [1]
#>  pROC            1.16.0     2020-01-12 [1]
#>  processx        3.4.1      2019-07-18 [1]
#>  prodlim         2019.11.13 2019-11-17 [1]
#>  promises        1.1.0      2019-10-04 [1]
#>  ps              1.3.0      2018-12-21 [1]
#>  purrr         * 0.3.3      2019-10-18 [1]
#>  R6              2.4.1      2019-11-12 [1]
#>  ranger          0.11.2     2019-03-07 [1]
#>  Rcpp            1.0.3      2019-11-08 [1]
#>  recipes       * 0.1.9      2020-01-14 [1]
#>  remotes         2.1.0      2019-06-24 [1]
#>  reshape2        1.4.3      2017-12-11 [1]
#>  rlang           0.4.2      2019-11-23 [1]
#>  rmarkdown       2.0        2019-12-12 [1]
#>  rpart           4.1-15     2019-04-12 [1]
#>  rprojroot       1.3-2      2018-01-03 [1]
#>  rsample       * 0.0.5      2019-07-12 [1]
#>  rsconnect       0.8.15     2019-07-22 [1]
#>  rstan           2.19.2     2019-07-09 [1]
#>  rstanarm        2.19.2     2019-10-03 [1]
#>  rstantools      2.0.0      2019-09-15 [1]
#>  rstudioapi      0.10       2019-03-19 [1]
#>  scales        * 1.1.0      2019-11-18 [1]
#>  sessioninfo     1.1.1      2018-11-05 [1]
#>  shiny           1.4.0      2019-10-10 [1]
#>  shinyjs         1.0        2018-01-08 [1]
#>  shinystan       2.5.0      2018-05-01 [1]
#>  shinythemes     1.1.2      2018-11-06 [1]
#>  SnowballC       0.6.0      2019-01-15 [1]
#>  StanHeaders     2.19.0     2019-09-07 [1]
#>  stringi         1.4.5      2020-01-11 [1]
#>  stringr         1.4.0      2019-02-10 [1]
#>  survival        2.44-1.1   2019-04-01 [1]
#>  testthat        2.3.1      2019-12-01 [1]
#>  threejs         0.3.1      2017-08-13 [1]
#>  tibble        * 2.1.3      2019-06-06 [1]
#>  tidymodels    * 0.0.3.9000 2019-12-11 [1]
#>  tidyposterior   0.0.2      2018-11-15 [1]
#>  tidypredict     0.4.3      2019-09-03 [1]
#>  tidyr         * 1.0.0      2019-09-11 [1]
#>  tidyselect      0.2.5      2018-10-11 [1]
#>  tidytext        0.2.2      2019-07-29 [1]
#>  timeDate        3043.102   2018-02-21 [1]
#>  tokenizers      0.2.1      2018-03-29 [1]
#>  tune          * 0.0.1      2020-01-14 [1]
#>  usethis         1.5.1      2019-07-04 [1]
#>  utf8            1.1.4      2018-05-24 [1]
#>  vctrs           0.2.1      2019-12-17 [1]
#>  withr           2.1.2      2018-03-15 [1]
#>  workflows     * 0.1.0.9000 2020-01-14 [1]
#>  xfun            0.12       2020-01-13 [1]
#>  xtable          1.8-4      2019-04-21 [1]
#>  xts             0.11-2     2018-11-05 [1]
#>  yaml            2.2.0      2018-07-25 [1]
#>  yardstick     * 0.0.4      2019-08-26 [1]
#>  zeallot         0.1.0      2018-01-28 [1]
#>  zoo             1.8-6      2019-05-28 [1]
#>  source                                
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.1)                        
#>  Github (r-lib/cli@fc1a494)            
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  Github (tidyverse/glue@c02d7d4)       
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  Github (tidymodels/hardhat@da3e5ef)   
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  Github (yihui/knitr@33d69c3)          
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  Github (tidymodels/modeldata@aa91bb1) 
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  Github (tidymodels/parsnip@1c2612a)   
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  Github (tidymodels/recipes@5e7c702)   
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.1)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  Github (tidymodels/tidymodels@4a9a694)
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  Github (tidymodels/tune@af40863)      
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  Github (tidymodels/workflows@c89bc0c) 
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#>  CRAN (R 3.6.0)                        
#> 
#> [1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
```

</details>

**Comment by kdpsingh** [comment]
I wonder adding a step_clean_names() function to recipes would help resolve issues like this (borrowing from the janitor::make_clean_names function). This could be the last item in the pipeline and would replace spaces with underscores and otherwise help enforce consistent case. Just an idea.

**Comment by apreshill** [comment]
For me, the central issue here is that this is NOT about column names, which `ranger` would indeed fuss about and throw an error if there were spaces in the column names.

The issue is that, under the hood, because I'm using a formula rather than a recipe, `hardhat` is converting my factor variables to dummy variables. So, my variable called Country has **values** that include spaces, which are in turn converted to names of dummy variables, and **then** `ranger` throws an error.

But to be clear, by using a formula, I was trying to not do any pre-processing. If I ran this model with the ranger package it would work. The issue is that hardhat is dummifying my predictor here. It is counter-intuitive to need to wrap my formula up in a recipe to *stop* this pre-processing from taking place. This is also why the warning I'm getting is impossible to de-bug without knowing this line of code to add before fitting the model:

```
blueprint = hardhat::default_formula_blueprint(indicators = FALSE)
```

**Comment by juliasilge** [comment]
This is closed in tidymodels/parsnip#332 and tidymodels/workflows#53:

``` r
library(tidymodels)
#> ── Attaching packages ─────────────────── tidymodels 0.1.0 ──
#> ✓ broom     0.5.6          ✓ recipes   0.1.13    
#> ✓ dials     0.0.7.9000     ✓ rsample   0.0.7     
#> ✓ dplyr     1.0.0          ✓ tibble    3.0.1     
#> ✓ ggplot2   3.3.2          ✓ tune      0.1.0.9000
#> ✓ infer     0.5.2          ✓ workflows 0.1.1.9000
#> ✓ parsnip   0.1.1.9000     ✓ yardstick 0.0.6     
#> ✓ purrr     0.3.4
#> ── Conflicts ────────────────────── tidymodels_conflicts() ──
#> x purrr::discard() masks scales::discard()
#> x dplyr::filter()  masks stats::filter()
#> x dplyr::lag()     masks stats::lag()
#> x recipes::step()  masks stats::step()
library(modeldata)

data(stackoverflow)

so_split <- initial_split(stackoverflow)
so_train <- training(so_split)

so_fold <- mc_cv(so_train, times = 1)

bag_spec <- 
  rand_forest() %>% 
  set_engine("ranger") %>% 
  set_mode("classification")

bag_spec %>%
  fit_resamples(
  Remote ~ Country,
  resamples = so_fold 
)
#> # Resampling results
#> # Monte Carlo cross-validation (0.75/0.25) with 1 resamples  
#> # A tibble: 1 x 4
#>   splits            id        .metrics         .notes          
#>   <list>            <chr>     <list>           <list>          
#> 1 <split [3.1K/1K]> Resample1 <tibble [2 × 3]> <tibble [0 × 1]>
```

<sup>Created on 2020-07-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

The factor variable is no longer surprisingly 😱  changed to indicator variables.
